components/browser_ui/media/android/java/src/org/chromium/components/browser_ui/media/BraveMediaSessionHelper.java

```diff
@@ -47,6 +43,15 @@ private boolean isBraveTalk() {
         return false;
     }
 
+    private boolean isBraveTalk() {
+        WebContents webContents =
+                (WebContents)
+                        BraveReflectionUtil.getField(
+                                MediaSessionHelper.class, "mWebContents", this);
+
+        return isBraveTalk(webContents);
+    }
```

Comment:
Reflective operations in Java are usually extremely slow. They can slow down the computation by 10x. 
What I would normally do is caching them in a static field so we don't access the reflection APIs multiple time.
So in this case it would be something like this:

```java
if (sWebContents == null) {
    sWebContents = (WebContents) BraveReflectionUtil.getField(MediaSessionHelper.class, "mWebContents", this);
}
return isBraveTalk(sWebContents);
```

Reply:
thaks, that's a good point. However I'm going to save it is a non static class member as that helper is get created different for each tab.

Reply:
actually, I see that upstream class have a set method for it https://github.com/chromium/chromium/blob/main/components/browser_ui/media/android/java/src/org/chromium/components/browser_ui/media/MediaSessionHelper.java#L288 and I don't think we are 100% confident that it's not updated somewhere in the middle of our reflection calls to it. I would pick a safe approach and fetch it, we have 2 calls there ideally on a media page load, shouldn't slow down at all. It wouldn't be good if it calls it tens-hundreds times, but 2 calls per media page I think is ok.